core/channel.go

```diff
@@ -36,15 +37,19 @@ func (ch *Channel) GetType() *Type {
 }
 
 func (ch *Channel) Hash() uint32 {
-	return HashPtr(uintptr(unsafe.Pointer(ch)))
+	return ch.hash
 }
 
 func (ch *Channel) WithInfo(info *ObjectInfo) Object {
 	return ch
 }
 
-func MakeChannel(ch chan FutureResult) *Channel {
-	return &Channel{ch: ch}
+func MakeChannel(ch chan FutureResult, hash uint32) *Channel {
```

Comment:
Looks like all calls to `MakeChannel` pass `0` as `hash`, so this change doesn't seem meaningful to me. In general, here and in other places is seems odd to me that the hash value is passed as an argument rather than being derived from the "content" of an object itself. And mutable objects like channels and readers should have distinct hash values per instance anyways...

Reply:
The only reason for the extra argument is to remind any future Joker maintainers, who might add new calls, to strive to compute a more-stable hash than would be assigned by default.

I'm happy to remove those extra args, though, because it's so unlikely they'll ever be used (in the cases where all current callers pass 0).

Note that I think it's a tad "safer" to compute the hash value for a new object once and store it, as this PR does for a few Object types, than to compute repeatedly. My thinking is that, since it's an "unsafe" computation (from the pointer to the object), the existence of a hash value won't prevent the GC from moving the object if needed to allocate more memory (not sure Go's GC ever does this -- it'd have to adjust all legit pointers to the object). By memoizing the value once and storing it in the object, the hash value for the object won't change even if the object itself moves. Would be a fun bug to track down, albeit highly unlikely to ever happen....